Storage/ThirdParty/SwiftData.swift

```diff
@@ -417,7 +413,13 @@ public class SQLiteDBConnection {
     /// Returns a cursor pre-filled with the complete result set.
     func executeQuery<T>(sqlStr: String, factory: ((SDRow) -> T), withArgs args: [AnyObject?]? = nil) -> Cursor<T> {
         var error: NSError?
-        let statement = SQLiteDBStatement(connection: self, query: sqlStr, args: args, error: &error)
+        let statement: SQLiteDBStatement?
```

Comment:
We should clean this up and move the if let error unwrapping below inside the catch block and remove the error declaration since we don't need it anymore.